sources/packages/backend/apps/api/src/route-controllers/education-program-offering/education-program-offering.students.controller.ts

```diff
@@ -93,4 +104,44 @@ export class EducationProgramOfferingStudentsController extends BaseController {
       studyEndDate: offering.studyEndDate,
     };
   }
+
+  /**
+   * Gets the offering simplified details, not including, for instance,
+   * validations, approvals and extensive data.
+   * Useful to have an overview of the offering details, for instance,
+   * when the user needs need to have quick access to data in order to
+   * support operations like confirmation of enrolment or scholastic
+   * standing requests or offering change request.
+   * @param offeringId offering.
+   * @param purpose indicating the purpose to allow for the appropriate authorization flow to be used.
+   * @param studentUserToken student user token for authorization.
+   * @returns offering details.
+   */
+  @ApiUnauthorizedResponse({
+    description: "The student is not authorized for the provided offering.",
+  })
+  @Get("offering/:offeringId/summary-details")
+  async getOfferingSummaryDetailsById(
+    @Param("offeringId", ParseIntPipe) offeringId: number,
+    @Query("purpose", new ParseEnumQueryPipe(ApplicationOfferingPurpose))
+    purpose: ApplicationOfferingPurpose,
+    @UserToken() studentUserToken: StudentUserToken,
+  ): Promise<EducationProgramOfferingSummaryViewAPIOutDTO> {
+    if (
+      purpose === ApplicationOfferingPurpose.OfferingChange &&
+      (await this.educationProgramOfferingControllerService.validateApplicationOfferingForStudent(
+        offeringId,
+        studentUserToken.studentId,
+      ))
+    )
+      return this.educationProgramOfferingControllerService.getOfferingById(
```

Comment:
this method is throwing not found exception, pls add it to API error decorators
![image](https://github.com/bcgov/SIMS/assets/77353155/7d86ac6c-6ab4-4111-9536-e32fe538c8fa)